lib/smart_answer/state.rb

```diff
@@ -2,8 +2,12 @@
 
 module SmartAnswer
   class State < OpenStruct
-    def initialize(start_node)
-      super(current_node: start_node, path: [], responses: [], response: nil, error: nil)
+    def initialize(start_node, forwarding_responses: {})
+      super(current_node: start_node,
```

Comment:
That's a good shout I can see if that's not too painful to add as a final commit.

Reply:
I had a quick play with this and found it caused quite a lot of test churn. I'm going to open this as a separate PR that bases onto this one rather than bring that into this.

I explored whether it actually made more sense to have the node available since we seem to frequently access it from the flow object and it felt a bit more natural than storing a property (`current_node_name`).  Like many things in Smart Answers, applying it to the app was pretty painless and there was a meltdown across the test suite.

Reply:
This is up now: https://github.com/alphagov/smart-answers/pull/5407 - I couldn't get through the test problems quickly for it to return a node so went with `current_node_name`.